### PR TITLE
[IMP] mail: remove inheritance of composer on message

### PR DESCRIPTION
**Pull request by tde-banana-odoo**
Purpose
 * unnecessary inherit of composer on mail.message: lot of unused fields and
   message specific behavior (notify, check access) not relevant for
   composer;
 * simply duplicate fields to use and remove inheritance;

### CODE REVIEW

File: addons/mail/wizard/mail_compose_message.py [hunk 1]
```diff
@@ -101,55 +112,62 @@ def _get_composition_mode_selection(self):
...
     auto_delete_message = fields.Boolean('Delete Message Copy', help='Do not keep a copy of the email in the document communication history (mass mailing only)')
     template_id = fields.Many2one(
         'mail.template', 'Use template', index=True,
         domain="[('model', '=', model)]")
-    # mail_message updated fields
-    message_type = fields.Selection(default="comment")
-    subtype_id = fields.Many2one(default=lambda self: self.env['ir.model.data'].xmlid_to_res_id('mail.mt_comment'))
-
-    @api.multi
-    def check_access_rule(self, operation):
```
Comment (Xavier-Do) [inline addons/mail/wizard/mail_compose_message.py]: LGTM exept for this, i'll check this tomorrow

File: addons/test_mail/tests/test_performance.py [hunk 2]
```diff
@@ -488,7 +488,7 @@ def test_complex_tracking_assignation(self):
         })
         self.assertEqual(rec.message_partner_ids, self.partners | self.env.user.partner_id)
 
-        with self.assertQueryCount(__system__=61, emp=76):  # com runbot: 61 - 76 // test_mail only: 60 - 72
+        with self.assertQueryCount(__system__=61, emp=76):  # com runbot: 61 - 76 // test_mail only: 61 - 73
```
[the remark this answers is not in the document]
Comment (tde-banana-odoo) [reply addons/test_mail/tests/test_performance.py]: Maybe it was not updated accordingly in master. Did not check on fresh instance, just updated all numbers.

**Comment by Xavier-Do** [comment]
robodoo r+

**Comment by Xavier-Do** [comment]
robodoo r+ rebase-ff

File: addons/mail/wizard/mail_compose_message.py [hunk 3]
```diff
@@ -101,55 +112,62 @@ def _get_composition_mode_selection(self):
...
+    res_id = fields.Integer('Related Document ID', index=True)
+    record_name = fields.Char('Message Record Name', help="Name get of the related document.")
+    # characteristics
+    message_type = fields.Selection([
+        ('comment', 'Comment'),
+        ('notification', 'System notification')],
+        'Type', required=True, default='comment',
+        help="Message type: email for email message, notification for system "
+             "message, comment for other messages such as user replies",
+        oldname='type')
```
Comment (KangOl) [inline addons/mail/wizard/mail_compose_message.py]: Why can't we use `email` as message_type using the wizard anymore (in 13.0)?

And the `help` message is not correct.

Comment (tde-banana-odoo) [reply addons/mail/wizard/mail_compose_message.py]: Because email is used for incoming emails.

About help, feel free to open a PR with your proposal.

**Pull request closed without merging**